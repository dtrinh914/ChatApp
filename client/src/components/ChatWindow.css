.ChatWindow{
    background: white;
    overflow: auto;
    flex-grow: 1;
    flex-shrink: 100;
}

.ChatWindow ul{
    margin: 0;
    padding: 0;
    list-style: none;
}

.ChatWindow li:nth-child(even){
    background: lightgrey;
}

.ChatWindow li{
    padding-left: 8px;
    word-break: break-all;
}