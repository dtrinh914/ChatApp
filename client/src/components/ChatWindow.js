import React from 'react';
import uuid from 'uuid/v4';
import Message from './Message';
import './ChatWindow.css';



function ChatWindow(props){
    return(
        <div className='ChatWindow'>
            <ul>
                {props.messages.map( message => <Message key={uuid()} text={message}/> )}
            </ul>
        </div>
    );
}

export default ChatWindow;