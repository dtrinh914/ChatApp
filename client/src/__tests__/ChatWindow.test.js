import React from 'react';
import ChatWindow from '../components/ChatWindow';
import {cleanup, getAllByTestId, render} from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

//seedData
const testMessages = [
                       {id: "5e28c5c185ffc434b337c266", 
                        text:"This is message 1.", 
                        time:"2020-01-22T22:02:35.511Z"
                       },
                       {id: "5e28c5bc85ffc434b337c264", 
                        text:"This is message 2.", 
                        time:"2020-01-22T22:03:36.511Z"
                       },
                       {id: "5e28c5c485ffc434b337c268", 
                        text:"This is message 3.", 
                        time:"2020-01-22T22:04:37.511Z"
                       }
                     ]
const testMemberMap = {
                        "5e28c5c185ffc434b337c266": {username:"test_user1"},
                        "5e28c5bc85ffc434b337c264": {username:"test_user2"},
                        "5e28c5c485ffc434b337c268": {username:"test_user3"}
                      }

afterEach(cleanup);

test('should match snapshot', ()=>{
    const {container} = render(<ChatWindow messages={testMessages} memberMap={testMemberMap}/>);
    expect(container).toMatchSnapshot();
});

test('should display three messages that is formated properly', ()=>{
    const {container} = render(<ChatWindow messages={testMessages} memberMap={testMemberMap}/>);
    const messages = getAllByTestId(container, 'message');

    expect(messages.length).toBe(3);
    expect(messages[0]).toHaveTextContent('This is message 1');
    expect(messages[0]).toHaveTextContent('test_user1');
    expect(messages[0]).toHaveTextContent('02:02 PM');

    expect(messages[1]).toHaveTextContent('This is message 2');
    expect(messages[1]).toHaveTextContent('test_user2');
    expect(messages[1]).toHaveTextContent('02:03 PM');

    expect(messages[2]).toHaveTextContent('This is message 3');
    expect(messages[2]).toHaveTextContent('test_user3');
    expect(messages[2]).toHaveTextContent('02:04 PM');
});

test('should have one properly formatted date divider', () =>{
    const {container} = render(<ChatWindow messages={testMessages} memberMap={testMemberMap}/>);
    const dateDividers = getAllByTestId(container, 'date-divider');

    expect(dateDividers.length).toBe(1);
    expect(dateDividers[0]).toHaveTextContent("Wed January 22, 2020");
});