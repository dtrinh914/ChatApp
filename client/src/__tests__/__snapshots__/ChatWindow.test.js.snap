// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
<div>
  <div
    class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <ul
      class="MuiList-root makeStyles-list-2 MuiList-padding"
    >
      <div
        class="makeStyles-root-35"
        data-testid="date-divider"
      >
        <p
          class="MuiTypography-root makeStyles-text-36 MuiTypography-body1"
        >
          Wed January 22, 2020
        </p>
        <hr
          class="MuiDivider-root makeStyles-divider-37"
        />
      </div>
      <li
        class="MuiListItem-root makeStyles-root-74 MuiListItem-gutters"
        data-testid="message"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-75 MuiAvatar-colorDefault"
        >
          T
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="username-time"
          >
            <strong>
              test_user1
            </strong>
            <span
              class="makeStyles-span-76"
            >
              02:02 PM
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="text-field"
          >
            This is message 1.
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-root-74 MuiListItem-gutters"
        data-testid="message"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-75 MuiAvatar-colorDefault"
        >
          T
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="username-time"
          >
            <strong>
              test_user2
            </strong>
            <span
              class="makeStyles-span-76"
            >
              02:03 PM
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="text-field"
          >
            This is message 2.
          </p>
        </div>
      </li>
      <li
        class="MuiListItem-root makeStyles-root-74 MuiListItem-gutters"
        data-testid="message"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-75 MuiAvatar-colorDefault"
        >
          T
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="username-time"
          >
            <strong>
              test_user3
            </strong>
            <span
              class="makeStyles-span-76"
            >
              02:04 PM
            </span>
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1"
            data-testid="text-field"
          >
            This is message 3.
          </p>
        </div>
      </li>
    </ul>
  </div>
</div>
`;
